src/main/java/io/singularitynet/utils/ImageClass.java

```diff
@@ -0,0 +1,804 @@
+package io.singularitynet.utils;
+
+import com.google.common.base.Charsets;
+import com.mojang.blaze3d.platform.GlConst;
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.TextureUtil;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.logging.LogUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Base64;
+import java.util.EnumSet;
+import java.util.Locale;
+import java.util.Set;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.client.util.Untracker;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.Nullable;
+import org.lwjgl.stb.STBIWriteCallback;
+import org.lwjgl.stb.STBImage;
+import org.lwjgl.stb.STBImageResize;
+import org.lwjgl.stb.STBImageWrite;
+import org.lwjgl.stb.STBTTFontinfo;
+import org.lwjgl.stb.STBTruetype;
+import org.lwjgl.system.MemoryStack;
+import org.lwjgl.system.MemoryUtil;
+import org.slf4j.Logger;
+
+@Environment(value=EnvType.CLIENT)
+public final class ImageClass
```

Comment:
Could you please add comment where this class comes from

Reply:
@noskill  this is a clone of NativeImage. It was cloned to change function writeTo so it now saves bmp not png bytes.

Reply:
I mean add comment to the source code.

Reply:
@noskill  done